modules/renter/repairscanner.go

```diff
@@ -0,0 +1,333 @@
+package renter
+
+// TODO: Make sure the `newMemory` channel for the renter is buffered out to one
+// element.
+
+// TODO: When building the chunk, need to also include a list of pieces that
+// aren't properly replicated yet.
+
+// TODO / NOTE: Once the filesystem is tree-based, instead of continually
+// looping through the whole filesystem we can add values to the file metadata
+// indicating the health of each folder + file, and the time of the last scan
+// for each folder + file, where the folder scan time is the least recent time
+// of any file in the folder, and the folder health is the lowest health of any
+// file in the folder. This will allow us to go one folder at a time and focus
+// on problem areas instead of doing everything all at once every iteration.
+// This should boost scalability.
+
+// TODO: If the original file would need to be downloaded to be repaired, don't
+// repair data that has less than 25% of its redundant pieces. (so in a
+// 10-of-110 situation, you repair when the piece count drops to 85 or lower).
+
+import (
+	"container/heap"
```

Comment:
👍 

maybe not a long-term solution, but makes more sense than rolling our own heap right now

Reply:
unlike with the tpool, we don't need special stuff in this heap, we can just use the generic heap.

Once we change the file mapping stuff, we will be dropping the heap anyway